macros/sql/surrogate_key.sql

```diff
@@ -0,0 +1,16 @@
+{% macro surrogate_key(fields) -%}
+
+md5(concat(
+
+    {%- for field in fields %}
+
+        coalesce(
+            {{dbt_utils.safe_cast(field, dbt_utils.type_string())}}
+            , '')
```

Comment:
I think we should concatenate each element in the list with `-`. Then, the number of items in the list is significant.

For instance, these two things should evaluate to different values:
```
{{ dbt_utils.surrogate_key(fields=['null']) }}

{{ dbt_utils.surrogate_key(fields=['null', 'null']) }}
```

In practice, this might happen with weirdness around utm parameters. Imagine:

| row | utm_medium  | utm_source |
| ---- | ------------- | ------------- |
| 1 | null | email  |
| 2 | email  | null  |

If we used the `surrogate_key` macro as it's currently implemented, both of these would evaluate to 

```
{{ dbt_utils.surrogate_key(fields=['utm_medium', 'utm_source']) }}

-->

md5('email')

```

whereas with dashes between each field, it would be:

```
-- Row 1
md5('-email')

-- Row 2
md5('email-')
```

Even the dash is not infallible here, but I think it's better than nothing!